components/eam/src/physics/cam/cloud_diagnostics.F90

```diff
@@ -284,6 +291,7 @@ subroutine cloud_diagnostics_calc(state,  pbuf)
 
     itim_old = pbuf_old_tim_idx()
     call pbuf_get_field(pbuf, cld_idx, cld, start=(/1,1,itim_old/), kount=(/pcols,pver,1/) )
+    call outfld('CLOUD',cld(:,:),pcols,lchnk)
```

Comment:
I guess I'm a little confused why the CLOUD outfld call was moved from cloud_cover_diags to here because it seems you are outputting the exact same variable.... Or was your intention to output "tot_cloud_frac" here instead of the E3SM "cld"?

Reply:
Previously, the CLOUD oufld call was done in the cloud_cover_diags_out subroutine. But since we are now using that same subroutine for either CLOUD or TOT_CLOUD_FRACTION, I didn't want TOT_CLOUD_FRACTION to overwrite CLOUD in the output. Does that make sense?

Reply:
Ah yes, Hassan is right. We call cloud_cover_diags_out below with *either* CLOUD or TOT_CLOUD_FRACTION as an input depending on whether we're using SHOC or not. So the variable CLOUD in cloud_cover_diags_out could refer to either the old or the new 3d cloud fraction. But we want to preserve the ability to write out both the old CLOUD (because that's the weirdly-defined cloud that the model actually uses) and also the new TOT_CLOUD_FRAC (because that's what will keep humans from rioting). So Hassan needed to move where the old CLOUD was output.